### PR TITLE
chore: add n-th Fibonacci number lab

### LINKED ISSUE #474: chore: add Nth fibonacci number dynamic programming lab to frontend cert(JS only)
There will be lectures videos to talk about dynamic programming.
But we need a lab for them to do it themselves.

A good problem would be to have a lab finding the Nth fibonacci number
https://www.geeksforgeeks.org/program-for-nth-fibonacci-number/

## Acceptance criteria

- [ ] create a new folder of the project in the `frontend-cert/js-projects` directory
- [ ] add your completed project in an `script.js` file in that new folder
- [ ] create a markdown file for your proposed user stories. You should call it `user-stories.md` and place it in your new project folder

### PR DESCRIPTION
**Pull request by Dario-DC**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #474

### CODE REVIEW

**Comment by zairahira** [comment]
It seems like the code is following a recursive approach rather than a dynamic approach. The goal of this lab isn't recursion according to the issue description.

**Comment by Dario-DC** [comment]
I refactored the code to use a bottom-up approach. This should count as dynamic @zairahira 

I'm asking to create an array in the global scope to store the computed sequence numbers because I thought it would be easier to test. I'm not sure, though.

File: frontend-cert/js-projects/nth-fibonacci-number/user-stories.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+1. You should define an empty array named `results` in the global scope.
+1. You should create a function named `fibonacci`.
+1. The `fibonacci` function should take single parameter, which is a positive integer number.
```
Comment (zairahira) [inline frontend-cert/js-projects/nth-fibonacci-number/user-stories.md:3]:
```suggestion
1. The `fibonacci` function should take a single parameter, which is a positive integer.
```

**Comment by zairahira** [comment]
This version implements the dynamic approach and it's looking good 🎉

**Review by zairahira — APPROVED** [review]

**Review by moT01 — COMMENTED** [review]
Pretty minor, but I would change the results variable to `sequence`.

I don't think it's working quite right, if I log `fibonacci(1)` I should get `0`, the first number in the sequence, but I get `1`. Here's some adjusted code:

```
const results = [0,1];

const fibonacci = (n) => {
    if (n <= 2) return results[n-1];

    for (let i = 2; i <= n; i++) {
        results.push(results[i - 1] + results[i - 2]);
    }
    return results[n-1]
}

console.log(fibonacci(10));
```

You don't have to use that, just an example.

**Comment by Dario-DC** [comment]
Good catch, I didn't realized it @moT01

**Comment by moT01** [comment]
The prototype doesn't fulfill the user stories anymore. "You should define an empty array named `sequence` in the global scope." We should either adjust the user story, or write a different solution that starts with an empty array.

**Comment by Dario-DC** [comment]
> The prototype doesn't fulfill the user stories anymore. "You should define an empty array named `sequence` in the global scope." We should either adjust the user story, or write a different solution that starts with an empty array.

My bad, I updated the name and forgot to fix the rest accordingly.

**Review by moT01 — APPROVED** [review]

**Pull request merged**